### PR TITLE
SQLite view introspection, flavours

### PR DESCRIPTION
**Pull request by pimeys**
Gotchas:

- Column arity is always optional for views, so we do the same trick in re-intro as we do with PostgreSQL.
- No default values in views.

Closes: https://github.com/prisma/prisma/issues/17416

### CODE REVIEW

File: introspection-engine/introspection-engine-tests/tests/simple.rs [hunk 1]
```diff
@@ -80,16 +80,29 @@ Example usage:
 source .test_database_urls/mysql_5_6
     "#);
 
+    if database_url == "sqlite" {
+        database_url = format!("{}.db", sqlite_test_url(test_function_name));
+
+        let file = database_url.trim_start_matches("file:");
+        std::fs::remove_file(&file).ok();
+    }
```
Comment (tomhoule) [inline introspection-engine/introspection-engine-tests/tests/simple.rs:88]: I think there's `sqlite_test_file()` in `test_setup` that does the sqlite file truncation too.

File: introspection-engine/connectors/sql-introspection-connector/src/pair/scalar_field.rs [hunk 2]
```diff
@@ -59,15 +59,9 @@ impl<'a> ScalarFieldPair<'a> {
...
-    /// we use the changed arity instead of the always optional value from the
-    /// database.
+    /// Optional, required or a list.
     pub fn arity(self) -> ColumnArity {
-        let keep_previous_arity = self.next.is_in_view()
-            && self.next.column_type().arity.is_nullable()
-            && self.context.sql_family.is_postgres();
-
-        if keep_previous_arity {
+        if self.context.flavour.keep_previous_scalar_field_arity(self.next) {
```
Comment (tomhoule) [inline introspection-engine/connectors/sql-introspection-connector/src/pair/scalar_field.rs:64]: :+1:

**Review by tomhoule — APPROVED** [review]
Awesome that we introduce flavours in introspection too, now. We can even use the same SqlFlavour from sql-migration-connector once we merge introspection-connector and migration-connector.

File: introspection-engine/introspection-engine-tests/tests/simple.rs (hunk 1 above)
Comment (pimeys) [reply introspection-engine/introspection-engine-tests/tests/simple.rs:88]: Actually, I'm already using this through `sqlite_test_url`. It somehow uses the same file in the subsequent runs... Therefore removing it here.

**Pull request merged**